scylla-cql/src/frame/types.rs

```diff
@@ -161,20 +162,22 @@ impl<'a> RawValue<'a> {
     }
 }
 
-fn read_raw_bytes<'a>(count: usize, buf: &mut &'a [u8]) -> Result<&'a [u8], ParseError> {
+fn read_raw_bytes<'a>(
+    count: usize,
+    buf: &mut &'a [u8],
+) -> Result<&'a [u8], LowLevelDeserializationError> {
     if buf.len() < count {
-        return Err(ParseError::BadIncomingData(format!(
-            "Not enough bytes! expected: {} received: {}",
-            count,
-            buf.len(),
-        )));
+        return Err(LowLevelDeserializationError::TooFewBytesReceived {
+            expected: count,
+            received: buf.len(),
+        });
     }
     let (ret, rest) = buf.split_at(count);
     *buf = rest;
     Ok(ret)
 }
 
-pub fn read_int(buf: &mut &[u8]) -> Result<i32, std::io::Error> {
+pub fn read_int(buf: &mut &[u8]) -> Result<i32, LowLevelDeserializationError> {
```

Comment:
This makes returned type less specific. If this function can only return io error then why force the caller to handle other errors?